### PR TITLE
Preparatory work for node-to-client protocol setup simplifications

### PR DESCRIPTION
**Pull request by dcoutts**
Preliminary (and related) simplifications before using
https://github.com/input-output-hk/ouroboros-network/pull/2056 to make
establishing a connection to a local node require much less configuration.

### CODE REVIEW

File: cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs [hunk 1]
```diff
@@ -747,13 +746,13 @@ pKESVerificationKeyFile =
       <> Opt.help "Filepath of the hot KES verification key."
       )
 
-pMaybeProtocolMagicId :: Parser (Maybe Byron.ProtocolMagicId)
-pMaybeProtocolMagicId =
-  optional $ Byron.ProtocolMagicId <$>
+pNetwork :: Parser Network
```
Comment (erikd) [inline cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs:749]: I am pretty sure you need `Parser (Maybe x)` for optparse to correctly detect that as an optional CLI parameter.

File: cardano-cli/src/Cardano/CLI/Common/Parsers.hs [hunk 2]
```diff
@@ -120,15 +120,15 @@ data ClientCommand
...
+        (Maybe SocketPath)
 
     -----------------------------------
 
   | SubmitTx
         TxFile
         -- ^ Filepath of transaction to submit.
         ConfigYamlFilePath
-        (Maybe CLISocketPath)
+        (Maybe SocketPath)
```
Comment (erikd) [inline cardano-cli/src/Cardano/CLI/Common/Parsers.hs:131]: :100:

File: cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs (hunk 1 above)
Comment (dcoutts) [reply cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs:749]: oops, I forgot so supply the default value of Mainnet.

**Comment by dcoutts** [comment]
bors merge

**Pull request merged**